kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/FromFileCreatable.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl;
+
+public interface FromFileCreatable<T> {
+  /**
+   * Create new resource from a directory or file contents.
+   *
+   * @param name resource name
+   * @param dirOrFilePath a file or directory path
+   * @return Resource instance for operations to do with this resource
+   */
+  T fromFile(String name, String... dirOrFilePath);
```

Comment:
I don't really like the fact that we are using Strings to specify files or paths.
What are the advantages of this vs. `T fromFile(Path path)` or `T fromFile(File file)`?

Reply:
Comparing to kubectl, the strings can actually be key file pairs - key=some.file